packml_plugin/include/packml_plugin/packml_widget.h

```diff
@@ -0,0 +1,159 @@
+/**
+ * @license Software License Agreement (Apache License)
+ *
+ * @copyright Copyright (c) 2017 Austin Deric
+ * @copyright Copyright (c) 2019 ROS-Industrial Consortium Asia Pacific (ROS 2 compatibility)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+#include <QtGui>
+#include <QtCore>
+#include <QtWidgets>
+#include "ui_packml.h"  // UI layout and components
+#include "rclcpp/rclcpp.hpp" // ROS 2 node
+#include "packml_msgs/srv/transition.hpp"  // Datatypes for packml topics and services
+#include "packml_msgs/srv/all_status.hpp"
+
+
+
+
+/**
+ * @brief This class creates a Widget (GUI) for an RViz plugin, to control and monitor a 
+ * PackML state machine
+ */
+class PackmlWidget : public QWidget
+{
+  Q_OBJECT
+public:
+
+
+  /**
+  * @brief Constructor of the class that creates the GUI plugin in Rviz, inherits a Widget
+  */
+  PackmlWidget(QWidget* parent = 0);
+
+
+  /**
+  * @brief Destructor for the widget object
+  */
+  virtual ~PackmlWidget();
+
+ 
+  /** 
+  * @brief Node for the Widget or RViz plugin
+  */
+  std::shared_ptr<rclcpp::Node> nh_;
+
+
+  /**
+  * @brief Creates the GUI layout with an UI object as reference
+  */
+  Ui::PackmlPanel* ui_;
```

Comment:
Could use a unique pointer here.

Reply:
Addressed in https://github.com/ros-industrial/packml_ros2/pull/2/commits/3c153847145310fca867ccce6de740feebf7f855